### ISSUE TITLE
[Feature] Please add more automatic generators

### ISSUE THREAD

**Issue by Moep90** [kapicorp/kapitan#694]
# Describe the feature
Currently only few [automatically](https://kapitan.dev/secrets/#automatically) generators are available.
## Automatically generate random integer
I can think of a `randomint` generator like the `randomstr`
In my case I'd like to generate random high userIDs for the security context.
For example:: `runAsUser: 1000620000 fsGroup: 1000620000`
Example:
```yaml
securityContext:
  runAsUser: ?{base64:targets/${target_name}/runAsUser||randomint:12}
```

## Ability to in/exclude special characters and/or 
If a tool only allows certain special characters I'd like to specify them.
I can think of something for example regex
Example:
```yaml
database_credentials:
  mysql_password_certain_special_chars: ?{base64:targets/${target_name}/runAsUser||randomstr("^[a-zA-Z0-9 ,:/\\-]*$"):6}
```

**Comment by MatteoVoges** [comment]
Actually, the issue is resolved with , but there is still a tiny problem:

The tags 
* `?{ref:path/to/secret||alphanumspec:16: }` - Error: whitespace as argument for special_chars (expected Non-Whitespace-Character)
* `?{ref:path/to/secret|randomstr}` - Error: using single `|` instead of `||`

get dumped in the output file and no secret was created.

I wan't at least that the error gets detected and a RefError gets raised, so that the user knows what he did wrong.

The new regex would detect both of it, but - i think - doesn't get evaluated.

Where do you think is the right place to check the tag for syntax first?

**Comment by Moep90** [comment]
closed by #866
